### PR TITLE
CI: fix subdirectory path globbing

### PR DESCRIPTION
**Pull request by anzz1**
- CI: Fix subdirectory path globbing
- CI: Fix Github runner AVX512F detection (Windows)

### CODE REVIEW

**Comment by j-f1** [comment]
IMO it would be reasonable to drop the paths and just always run CI, what do you think?

**Comment by anzz1** [comment]
> IMO it would be reasonable to drop the paths and just always run CI, what do you think?
Nah, it's definitely better this way. It's been working fine so far and will continue to work, just that change to include subdirectories too will restore it to working order.

It simply doesn't make any sense to run CI builds on every readme file change. Tbh someone should the fix the docker too that it only builds when it needs to. Before the docker was in the ci the PR's were very clean since the non-code changes didn't go through the check rigamarole.

Also there are security implications for having the workflow paths and push/pr types exactly like it currently is, it's not a random decision. The way I have set it up is so that I can condifently say that I've secured the logic reasonably well.

You do say reasonable, but I cannot think of any good reasons for it. Like I can think of many reasons more besides the ones already said why it's better to have the CI mindfully set up than having a *.* "just do everything everytime always lol" type of thing.

But I can't think of a single reason why the " *.* " CI build type would be better. I'm open to be convinced otherwise though, as maybe there is something I don't see or forget.

**Comment by j-f1** [comment]
> But I can't think of a single reason why the "do everything always check nothing" CI build type would be better

My rationale was that if the checks are always run, we can never miss a change that breaks CI. We’re not paying for the compute time to run the tests, so seeing the ✅ indicates that nothing broke, even if the change is just a readme change.

In terms of security, what are you worried about? As far as I can tell, nothing uses `pull_request_target`, fork PRs are not given any more access than they should have, and people with commit access to this repo can just do malicious things directly instead of having to use an action.

**Comment by anzz1** [comment]
> My rationale was that if the checks are always run, we can never miss a change that breaks CI. We’re not paying for the compute time to run the tests, so seeing the white_check_mark indicates that nothing broke, even if the change is just a readme change.

Ah I see, thats a valid point if the CI was misconfigured. However, when properly configured (meaning that anything that can affect CI is explicitly defined), there simply aren't any changes to break CI to do. Readme changes can't break CI.

> In terms of security, what are you worried about? As far as I can tell, nothing uses `pull_request_target`, fork PRs are not given any more access than they should have, and people with commit access to this repo can just do malicious things directly instead of having to use an action.

There are ways to cause mischief with the GH actions which I'm not going to go into more detail. You can "just trust me bro" or do your own research. 

It is also simply good practice to limit the scope of the CI actions and be mindful when configuring them. Various guides about working with actions go into more detail about this.

Yes, we are not paying for the compute but there are also the [Usage Limits](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration#usage-limits) to consider. And yes, we are not currently hitting any limits but I don't see why a code PR in the whisper.cpp should wait for a runner slot because maximum concurrent jobs are being reached because some readme changes were done in this repo at the same time. And even if we had all the unlimited compute in the world, I still don't get why argue for wasting energy even if it costs nothing to you.

I don't even understand why we are arguing about this? I can understand that there are times when doing something the "smart way" isn't worth the effort. But especially as the work to do it the "smart way" has already been put in, I don't get why it should be undone to do it the "dumb way".

Can we just get over this already and move on to better things? This is a dumb thing to have an argument about, we both have better things to do. Please :smile_cat:

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -8,10 +8,10 @@ on:
         required: true
         type: boolean
   push:
-    paths: ['.github/workflows/**', 'CMakeLists.txt', 'Makefile', '**.h', '*.c', '**.cpp']
+    paths: ['.github/workflows/**', '**/CMakeLists.txt', '**/Makefile', '**/*.h', '**/*.c', '**/*.cpp']
   pull_request:
     types: [opened, synchronize, edited, reopened, review_requested, ready_for_review]
-    paths: ['CMakeLists.txt', 'Makefile', '**.h', '*.c', '**.cpp']
+    paths: ['**/CMakeLists.txt', '**/Makefile', '**/*.h', '**/*.c', '**/*.cpp']
```
Comment (Green-Sky) [inline .github/workflows/build.yml:14]: someone explain to me why they look the way they do.

Comment (anzz1) [reply .github/workflows/build.yml:14]: It's the [glob format](https://en.wikipedia.org/wiki/Glob_(programming))

Comment (anzz1) [reply .github/workflows/build.yml:14]: In plain english it says:
in all directories and subdirectories
CMakeLists.txt
Makefile
*.h
*.c
*.cpp

Comment (Green-Sky) [reply .github/workflows/build.yml:14]: and you cant just use `**` instead of `**/*` ? that feels kind of wrong

**Comment by j-f1** [comment]
> Can we just get over this already and move on to better things? This is a dumb thing to have an argument about, we both have better things to do. Please 😸

Sure, I will not object to merging this PR. You make some good points!

File: .github/workflows/build.yml (hunk 1 above)
Comment (anzz1) [reply .github/workflows/build.yml:14]: > and you cant just use `**` instead of `**/*` ? that feels kind of wrong

Yeah, for example `**/Makefile` matches files named exactly `Makefile` , while `**Makefile` would also match `abcdefgMakefile` and so on. The `**/*` format is the proper format for recursing directories.

**Review by ggerganov — CHANGES_REQUESTED** [review]

**Review by ggerganov — APPROVED** [review]

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -177,7 +177,7 @@ jobs:
...
...4vh+JuL4fiai+P4m4si98aL4vibi7Xbq4vg+JuLUmljaOH4m4sAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAABQRQAATAEBAGmkImQAAAAAAAAAAOAADwELAQYAAAIAAAAAAAAAAAAADBAAAAAQAAAAIAAAAABAAAAQAAAAAgAABAAAAAAAAAAEAAAAAAAAAAAgAAAAAgAAAAAAAAMAAAAAABAAABAAAAAAEAAAEAAAAAAAABAAAAAAAAAAAAAAAFQQAAAoAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEAAADAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAC50ZXh0AAAAsgAAAAAQAAAAAgAAAAIAAAAAAAAAAAAAAAAAACAAAGAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAACUEAAAiBAAAAAAAABVi+xRUVNTM8m4BwAAAA+iwesQZoPjAYld/FuDRfwwjUX4agBQjUX8agFQavX/FQAQQABQ/xUEEEAAi0X8W4PA0PfYG8BAycN8EAAAAAAAAAAAAACkEAAAABAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAlBAAAIgQAAAAAAAApANXcml0ZUZpbGUAuQFHZXRTdGRIYW5kbGUAAEtFUk5FTDMyLmRsbAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA==')) -AsByteStream
```
[the remark this answers is not in the document]
Comment (anzz1) [reply .github/workflows/build.yml]: New implemenation, replacing opacity with verbosity #584
